exts/betterTooltip/betterTooltip.js

```diff
@@ -0,0 +1,537 @@
+/*!
+ * Copyright 2002 - 2016 Webdetails, a Pentaho company.  All rights reserved.
+ *
+ * This software was developed by Webdetails and is provided under the terms
+ * of the Mozilla Public License, Version 2.0, or any later version. You may not use
+ * this file except in compliance with the license. If you need a copy of the license,
+ * please go to  http://mozilla.org/MPL/2.0/. The Initial Developer is Webdetails.
+ *
+ * Software distributed under the Mozilla Public License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or  implied. Please refer to
+ * the license for the specific language governing your rights and limitations.
+ */
+
+(function() {
+    "use strict";
+
+    /*global define:true, pvc:true, def:true */
+
+    function moduleDef(def, pvc) {
+
+        /**
+         * @class
+         * @name pvc.ext.BetterTooltip
```

Comment:
I'd vote for something related with the characteristics of the tooltip.
The base `categoryDetailsTooltip` name is due to it displaying the details of the current category. It shows the value of a measure in each of the existing series.
This tooltip displays information only about the current visual element: its category, series and measure(s).
Then, this is not a generic tooltip: it is designed to display only certain well-known visual/roles of a chart. Even with configuration, it will not (or will, hardly) scale to be used with any chart type. The existing, default CCC tooltip is generic, in that sense.

This tooltip displays data organized by visual role
On the other hand, the CCC tooltip lists the value of each underlying CCC data dimension, annotating each with a symbol that related with the visual roles it is playing.
However, this tooltip is not a generic by-visual-role tooltip, as it focuses on specific visual roles (or classifications of):  category, series/color, measure(s).

The closest to its function that I can come up with, albeit still a bit over, is "categoricalTooltip". Category and series are both seen as _categories_.

A more specific name would be "twoLevelCategoricalTooltip", evidencing the fact that there are two types of categories being handled. Although this doesn't look good and unnecessarily limits its future scope.

As to adding a year suffix, if we change the name now, we might not need the year, and only add the year if the same design structure is reiterated with a new style next year :-)

I don't appreciate the names: BetterTooltip or Tooltip2016.